Fix logical mistakes for ARC

qwe661234 opened the pull request:
After importing more cache tests, we discovered some logical mistakes in replacing cache entries. This commit patches these mistakes.

qwe661234 committed:
Add test case for cache test

jserv commented on src/cache.c:
```diff
@@ -249,6 +249,20 @@ cache_t *cache_create(int size_bits)
                cache->list_size[LFU_ghost_list] <=                          \
            2 * cache->capacity);
 
+#define replace_list(op1, op2)                                                 \
```
Capitalize this macro.

qwe661234 committed:
Fix logical mistakes for ARC

After importing more cache tests, we discovered some logical mistakes in
replacing cache entries. This commit patches these mistakes.
